pkg/server/skydb/recordutil/recordutil.go

```diff
@@ -328,21 +328,33 @@ func RecordSaveHandler(req *RecordModifyRequest, resp *RecordModifyResponse) sky
 	makeAssetsCompleteAndInjectSigner(db, req.Conn, records, req.AssetStore)
 
 	// execute after save hooks
-	if req.HookRegistry != nil {
-		records = executeRecordFunc(records, resp.ErrMap, func(record *skydb.Record) (err skyerr.Error) {
-			originalRecord, _ := originalRecordMap[record.ID]
-			err = req.HookRegistry.ExecuteHooks(req.Context, hook.AfterSave, record, originalRecord)
-			if err != nil {
+	records = hookTriggerer.triggerIfRegistered(records, hook.AfterSave, true)
+
+	resp.SavedRecords = records
+	resp.SchemaUpdated = schemaExtended
+
+	return nil
+}
+
+type saveHookTriggerer struct {
+	Req               *RecordModifyRequest
+	Resp              *RecordModifyResponse
+	OriginalRecordMap map[skydb.RecordID]*skydb.Record
+}
+
+func (t saveHookTriggerer) triggerIfRegistered(records []*skydb.Record, kind hook.Kind, shouldLogErr bool) []*skydb.Record {
```

Comment:
> All code does or does not do something based on condition

All code does `something` or does `something else` based on condition, but not all are `does or does not do something`.

You may argue that the `xxxIfxxx` is not consistent in the project or this kind of function is not a good practice, which can be a reason to refactor the function. But IMO if a function has the behaviour that `it does {A} if {B}, or it does nothing if not {B}`, its name should be `{A}if{B}`. Otherwise, it is a trap which makes other developers to think that this function would always do {A}.